inc/saiicmpecho.h

```diff
@@ -0,0 +1,387 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiicmpecho.h
+ *
+ * @brief   This module defines SAI interface
+ */
+
+#if !defined (__SAIICMPECHO_H_)
+#define __SAIICMPECHO_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIICMPECHO SAI - ICMP_ECHO specific public APIs and data structures
+ *
+ * @{
+ */
+
+/**
+ * @brief SAI ICMP_ECHO session state
+ */
+typedef enum _sai_icmp_echo_session_state_t
+{
+    /** ICMP_ECHO Session is in Down */
+    SAI_ICMP_ECHO_SESSION_STATE_DOWN = 0,
+
+    /** ICMP_ECHO Session is Up */
+    SAI_ICMP_ECHO_SESSION_STATE_UP,
+
+} sai_icmp_echo_session_state_t;
+
+/**
+ * @brief Defines the operational status of the ICMP_ECHO session
+ */
+typedef struct _sai_icmp_echo_session_state_notification_t
+{
+    /**
+     * @brief ICMP_ECHO Session id
+     *
+     * @objects SAI_OBJECT_TYPE_ICMP_ECHO_SESSION
+     */
+    sai_object_id_t icmp_echo_session_id;
+
+    /** ICMP_ECHO session state */
+    sai_icmp_echo_session_state_t session_state;
+
+} sai_icmp_echo_session_state_notification_t;
+
+/**
+ * @brief SAI attributes for ICMP_ECHO session
+ */
+typedef enum _sai_icmp_echo_session_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_START,
+
+    /**
+     * @brief Hardware lookup valid
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default true
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_HW_LOOKUP_VALID = SAI_ICMP_ECHO_SESSION_ATTR_START,
```

Comment:
I believe hw lookup means that ICMP packet is forwarded using L3 lookup instead of forwarding to a port. Please rename the attribute and add comments to that intent.

Reply:
You are correct. It's intended for L3 lookup for forwarding. I am willing to rename it, but the same naming convention is also used in saitwamp.h and saibfd.h, and I prefer to maintain consistency.  Please let me know if still want me to rename it.

Reply:
In that case please add details in comments clearly stating what happens when flag is set to true or false

Reply:
Added a brief description for this attribute.  Also, this attribute is mentioned explained in the examples.

Reply:
Thank You